src/breakpoint/SetHWBreakpoint.ts

```diff
@@ -0,0 +1,78 @@
+import { ExtensionContext, window } from 'vscode';
+import * as vscode from 'vscode';
+import { BreakpointStatusBarItem } from './BreakpointStatusBarItem';
+import * as path from 'path';
+
+export class SetHWBreakpoint {
+    constructor(context: ExtensionContext) {
+        this.registerCommands(context);
+    }
+
+    private readonly registerCommands = (context: ExtensionContext) => {
+        const breakpointStatusBarItem = new BreakpointStatusBarItem();
+        context.subscriptions.push(breakpointStatusBarItem);
+        vscode.debug.onDidChangeActiveDebugSession(async (debugSession) => {
+            if (debugSession) {
+                const workspaceFolder = vscode.workspace.workspaceFolders?.[0];
+                if (!workspaceFolder) {
+                    throw new Error('No workspace folder found.');
+                }
+                const configPath = path.join(
+                    workspaceFolder.uri.fsPath,
+                    '.vscode',
+                    'launch.json'
+                );
+                const config: vscode.WorkspaceConfiguration =
+                    vscode.workspace.getConfiguration(
+                        'launch',
+                        vscode.Uri.file(configPath)
+                    );
+                const HWBreakpointParameter: boolean | undefined =
+                    config?.configurations[0]?.hardwareBreakpoint;
```

Comment:
Is it possible to  get the current status from the backend rather than trying to maintain duplicate logic in the frontend?